mmdet/datasets/pipelines/transforms.py

```diff
@@ -2099,16 +2103,25 @@ def _mosaic_transform(self, results):
 
         if len(mosaic_labels) > 0:
             mosaic_bboxes = np.concatenate(mosaic_bboxes, 0)
-            mosaic_bboxes[:, 0::2] = np.clip(mosaic_bboxes[:, 0::2], 0,
-                                             2 * self.img_scale[1])
-            mosaic_bboxes[:, 1::2] = np.clip(mosaic_bboxes[:, 1::2], 0,
-                                             2 * self.img_scale[0])
             mosaic_labels = np.concatenate(mosaic_labels, 0)
 
+            if self.bbox_clip_border:
+                mosaic_bboxes[:, 0::2] = np.clip(mosaic_bboxes[:, 0::2], 0,
+                                                 2 * self.img_scale[1])
+                mosaic_bboxes[:, 1::2] = np.clip(mosaic_bboxes[:, 1::2], 0,
+                                                 2 * self.img_scale[0])
+
             if not self.skip_filter:
                 mosaic_bboxes, mosaic_labels = \
                     self._filter_box_candidates(mosaic_bboxes, mosaic_labels)
 
+        # remove outside bboxes
+        inside_inds = remove_outside_bboxes(mosaic_bboxes,
```

Comment:
The original logic of this operation does not exist, and it is unclear whether it will affect performance.

Reply:
In the original logic, all `bboxes` are clipped, which means that all `bboxes` are inside the image. 
Therefore, there is no difference after `remove_outside_bboxes`.

Reply:
The `remove_outside_bboxes` only remove `bboxes` that are completely out of the image boundary.
The `bbox` will be kept as long as a part of it is inside the image.

Reply:
The clip may have data like [0,0,2,2], but I think it’s okay to filter it out.